transport/controller.cc

```diff
@@ -212,7 +212,9 @@ future<> controller::do_start_server() {
             });
         }).get();
 
-        set_cql_ready(true).get();
+        if (!_used_by_maintenance_socket) {
+            set_cql_ready(true).get();
+        }
```

Comment:
Maybe a more correct approach is to require that all cql servers are ready. If port 9042 is ready but port 19042 is not, then we aren't ready.

Reply:
We start listening on all ports in the block above: `parallel_for_each(configs, ...`